StaticInstaller.php

```diff
@@ -123,7 +155,94 @@ public static function composerFinishUpdate(Event $composer_event = null)
+            }
+            self::printToStdout("Delete unused ".$app_alias." app backup\n");
+        }
+        catch(\Exception $e){
+            self::printToStdout("Could not delete ".$app_alias.". ".$e->getMessage() . ' in ' . $e->getFile() . ' at line ' . $e->getLine() . ".\n");
+            return false;
+        }
+        return true;
+    }
+    /**
+     * Backup all apps that are listen in composer.json but not listed in EXCLUDE_BACKUP_PACKAGES
+     * since they have no own backup function to call to
+     *
+     * @param Event $composer_event
+     * @return Event $composer_event
+     */
+    public static function composerBackupEverything(Event $composer_event = null){
+        $installer = new self();
+        $apps = $installer->getAllApps();
+         //write consistent backuptime to delete excess data after update run
+        $backupTime = date('Y_m_d_H_i');
+        $temp = self::getTempFile();
+        $temp['backupTime'] = $backupTime;
+        self::setTempFile($temp);
+        foreach( $apps as $app){
+            if (!in_array($app,self::EXCLUDE_BACKUP_PACKAGES)){
+                $installer->backup($app, $backupTime);
+            }
+        }
+        return $composer_event;
+    }
 
+    public function getAllApps(){
+        $ds = DataSheetFactory::createFromObjectIdOrAlias($this->getWorkbench(), 'exface.Core.APP');
+        $result = $ds->dataRead();
+        $row = $ds->getRows();
+        $apps = array();
+        foreach ($row as $fld => $val) {
+            $app = $this->getWorkbench()->getApp($val['UID']);
+            $apps[] = $app->getAliasWithNamespace();
+        }
+        return $apps;
+    }
+    /**
+     * Call backup function on app, install at specified backup folder, folder name is defined by backupTime-String
+     * @param string $app_alias
+     * @param string $backupTime
+     * @return string
+     */
+    public function backup($app_alias, $backupTime){
+        $exface = $this->getWorkbench();
+        $result = '';
+        try {
+            $text = '-> Creating backup of app ' . $app_alias . "\n";
+            self::printToStdout($text);
+            $app_name_resolver = NameResolver::createFromString($app_alias, NameResolver::OBJECT_TYPE_APP, $exface);
+            $backupAction = $exface->getApp(self::PACKAGE_MANAGER_APP_ALIAS)->getAction(self::PACKAGE_MANAGER_BACKUP_ACTION_ALIAS);
+            $backupAction->setBackupPath("autobackup".DIRECTORY_SEPARATOR.$backupTime);
+            $backupAction->backup($app_name_resolver);
+        }
+        catch(\Exception $e){
+            if ($e instanceof ExceptionInterface){
+                $log_hint = ' (see log ID ' . $e->getId() . ')';
+            }
+            self::printToStdout('-> Error in ' . $app_alias . "! \n");
+            self::printToStdout($e->getMessage() . ' in ' . $e->getFile() . ' at line ' . $e->getLine() . $log_hint."\n");
+            //$exface->getLogger()->logException($e);
```

Comment:
Warum loggst du die Exception nicht? Wenn es da Fehler gab, sprich dazu bitte mit Stefan. Er hat gerade heute einen Ähnlichen Fall gehabt.

Installer-Fehler gehören immer in den Log. In der Ausgabe sind keine Details enthalte und solche Fehler sind schwer zu reproduzieren! Zumal du ja schon eine Log-ID ausgibst (korrekt) - nur schreibst du keinen Log-Eintrag.

Reply:
War noch von einem vorherigen Problem auskommentiert, ist jetzt drin.